src/State/helpers.js

```diff
@@ -25,3 +27,92 @@ export const transformRequests = (requests, role) => {
     }
   });
 };
+
+/* constructs a request config, then passing that to aws4 to sign
+ * host: the base url of the API
+ * method: the HTTP method of the request
+ * path: the specific path to be appended to the host (e.g. /users/requests)
+ * params: any query paramters the request needs. Pass in null if there is no query string parameters
+ * data: the body of the request. Pass in null if there is no body
+ */
+export const constructRequest = (host, method, path, params, data) => {
+  const {
+    AccessKeyId,
+    EmpEmail,
+    SamlHash,
+    SecretKey,
+    SessionToken,
+    UserKey
+  } = store.getState().currentUser;
+
+  let userSession = {
+    AccessKeyId,
+    EmpEmail,
+    SamlHash,
+    SecretKey,
+    SessionToken,
+    UserKey
+  };
+
+  // construct request
+  let request = {
+    host,
+    method,
+    url: `${host}${path}`,
+    path,
+    data: { ...userSession }, // data paramter needed for axios
+    body: JSON.stringify({ ...userSession }) // body parameter needed for aws
```

Comment:
Do you need to stringify since we are using `axios`

Reply:
`axios` reads from the `data` param, which you are correct doesn't need to be stringified.  However, since `aws4` uses `http.request` behind the scenes, it doesn't take in a `data` param, but rather a `body` param that needs to be stringifed. Hence, I pass in both. The sources I linked in this MR explains this in more detail

Reply:
Sorry I didnt have a change to read the link. Just read them now. Pretty good